File: 동적계획법/leetcode1137(Easy)-N-th Tribonacci Number.js
// 69ms(51.56%), 42MB(24.68%)
/**
 * @param {number} n
 * @return {number}
 */
var tribonacci = function (n) {
  const dp = [0, 1, 1]
  for (let i = 3; i < n + 1; i++) {
    const t = dp[i - 3] + dp[i - 2] + dp[i - 1]
    dp.push(t)
  }
  return dp[n]
}

File: 동적계획법/leetcode1137(Easy)-N-th Tribonacci Number.py
# 39ms(39.1%), 13.8MB(54.1%)
class Solution:
    def tribonacci(self, n: int) -> int:
        dp = [0, 1, 1]
        for i in range(3, n + 1):
            t = dp[i - 3] + dp[i - 2] + dp[i - 1]
            dp.append(t)
        return dp[n]